crates/cli-flags/src/lib.rs

```diff
@@ -532,6 +542,23 @@ impl CommonOptions {
             strategy => config.strategy(strategy),
             _ => err,
         }
+        match_feature! {
+            ["gc" : self.codegen.collector]
+            collector => match collector {
+                #[cfg(not(feature = "gc-drc"))]
+                wasmtime::Collector::DeferredReferenceCounting => anyhow::bail!(
+                    "support for the `drc` collector is unavailable because the \
+                     `gc-drc` feature was disabled at compile time",
+                ),
+                #[cfg(not(feature = "gc-null"))]
+                wasmtime::Collector::Null => anyhow::bail!(
+                    "support for the `null` collector is unavailable because the \
+                     `gc-null` feature was disabled at compile time",
+                ),
```

Comment:
Could this get pushed into a `Result`-return of `Config::collector` or within `Config::validate`?